ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh

```diff
@@ -37,7 +37,16 @@ cp "${SHARED_DIR}/install-config.yaml" "${dir}/"
 mkdir -p ~/.ssh
 cp "${SSH_PRIV_KEY_PATH}" ~/.ssh/
 
-openshift-install --dir="${dir}" create manifests &
+if [[ "${USE_ETCD_RAMDISK}" == "true" ]]; then
```

Comment:
If this is something that needs to be done always for specific platforms, can we keep that logic isolated in the step rather than introducing a separate variable to the step? In other words, check ${CLUSTER_TYPE} here rather than "${USE_ETCD_RAMDISK}.

Reply:
This is really not platform specific. It is "cloud provider" specific I think.  It is necessary if the "infra" being used has performance issues. 
For example, it is prevelant on the MOC openstack cloud provider, but not so much in the VEXHOST openstack cloud provider. 
MOC and VEXHOST are two different vendors that we use to test openstack deployments. 
I imagine is a similar situation for kubevirt and ovirt, depending on what is used to run the ci test, the hardware being used might have problems that cause ci tests to fail when the systems are being stressed. for example when many ci tests are running simultatneously on that hardware. 

we could limit it to a particular platform in this file but in reality that is the same thing as using the USE_ETCD_RAMDISK environmental variable. 
It's just that instead of checking the platform here, it is set per platform on the test definition (workflow) which will allow platform to test both cases: with and without ramdisk.  You can see an exampl of that in the file rator/step-registry/openshift/e2e/openstack/ipi/openshift-e2e-openstack-ipi-workflow.yaml which is also being edited as part of this patch. 
I really don't think is a good idea to put too much platform dependent logic into this step. Is already comboluted with checking all different platforms.. There should not really be any platform check in this step, but that ship has sailed already. But we should not add more fuel to the fire.

Reply:
Yeah, you are right. It is better to have the USE_ETCD_RAMDISK variable.

I still have concerns over testing using an unsupported configuration. I may not fully understand the scope of the problem that you are trying to solve, but I have a hard time believing that the risk of testing with an unsupported configuration outweighs the burden of test failures when the test infrastructures is being stressed. Ultimately, it's your team's decision whether you want to accept that risk. I suspect that other teams are not going to be inclined to help you debug issues that arise in your CI tests when they discover that the etcd is running on RAM disks.

Reply:
I think we'll be ok in our team. We already run our tests with ramdisk,(templated) and actually the reason we are looking at adding this to step registry is because we are trying to migrate from the template to step registry. 
Adding it with the USE_ETCD_RAMDISK just makes it so we don't all have to keep different versions of ipi-install-install.

Reply:
Perhaps a different step may be required. 
we already run our templated tests with ramdisks. The reason for this PR is because we were trying to migrate to step registry. 
The goal we had was to be able to use the ipi-install-install step but in order to do so we need to make certain modifications like this one. 

If we are the only platform that requires this, then it might make sense to have our own "openstak-ipi-install" step, but it's my understanding that a couple more platforms require it as well, which would mean a total of four steps (one under each platform step registry directories,  that basically do the same thing
We might end up with half a dozen different ipi-install-install and that might be hard to mantain.

Reply:
If you are comfortable with using ramdisk (and already use it anyway), then I won't stand in the way of these changes.

Reply:
I think using the USE_ETCD_RAMDISK env as a gatekeepr (default to false) would make this change safe for other platforms workflows.